src/IR.cpp

```diff
@@ -913,6 +913,15 @@ Stmt Atomic::make(const std::string &producer_name,
     return node;
 }
 
+Stmt HoistedStorage::make(const std::string &name,
+                          Stmt body) {
+    HoistedStorage *node = new HoistedStorage;
+    node->name = name;
+    internal_assert(body.defined()) << "HoistedStorage must have a body statement.\n";
```

Comment:
The assert should be moved up to the top of the function, because internal assert might throw, someone might catch, and we don't want to leak HoistedStorage in that case.